### ISSUE TITLE
Is `KItem ~> .(.KList)` even possible?

### ISSUE THREAD

**Issue by grosu** [kframework/k-legacy#2016]
Very strange behavior with 1_lambda\lesson_2.

```
D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_2>kompile lambda.k
[Warning] Compiler: Could not find main syntax module with name LAMBDA-SYNTAX
in definition.  Use --syntax-module to specify one. Using LAMBDA as default.
D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_2>krun ..\lesson_1\tests\closed-variable-capture.lambda
<k> ( lambda y . ( ( lambda x . lambda y . ( x y ) ) y ) ) ~> . </k>
```

What is the trailing `.`? Going further:

```
D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_2>krun ..\lesson_1\tests\closed-variable-capture.lambda --
output kast
`<k>`(#KSequence(`lambda_._`(#token("y","Id"),`__`(`lambda_._`(#token("x","Id"),`lambda_._`(#token("y","Id"),`__`(#token
("x","Id"),#token("y","Id")))),#token("y","Id"))),#EmptyK(.KList)))
```

So it looks like `.::K` is a KLabel ... Is that intended?  That is certainly different from the theory of K.

On the other hand, search with a pattern works fine:

```
D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_2>krun ..\lesson_1\tests\closed-variable-capture.lambda --search --pattern "<k>V:K</k>"
Search results:

Solution 1:
V:K -->
lambda y . ( ( lambda x . lambda y . ( x y ) ) y )

D:\github\grosu\k\k-distribution\tutorial\1_k\1_lambda\lesson_2>krun ..\lesson_1\tests\closed-variable-capture.lambda --search --pattern "<k>V:K</k>" --output kast
Search results:

Solution 1:
V:K -->
`lambda_._`(#token("y","Id"),`__`(`lambda_._`(#token("x","Id"),`lambda_._`(#token("y","Id"),`__`(#token("x","Id"),#token
("y","Id")))),#token("y","Id")))
```

@cos, @andreistefanescu ?

**Comment by cos** [comment]
at parsing time, yes, `.::K` is a label -- @radumereuta knows why this is necessary

**Comment by cos** [comment]
The extra `~> .` is an artifact of the backend implementation. If it's not that bad, I would leave it alone for now.

**Comment by andreistefanescu** [comment]
The `~> .` is removed after the assoc pull request.

**Comment by grosu** [comment]
I prefer what @andreistefanescu said, because my suspicion is that it will resurface in other definitions.  It may work here because of search with a pattern, but who knows what happens in FUN or other languages that we have not translated yet.

**Comment by andreistefanescu** [comment]
Fixed in #1955 .
